$OpenBSD: patch-src_libical_icalerror_h,v 1.2 2011/11/14 23:55:48 ajacoutot Exp $

Define ICAL_ERRORS_ARE_FATAL to unbreak ports using -Werror while
including ical.h since on OpenBSD undefined declarations will produce a
warning:
https://sourceforge.net/tracker/?func=detail&aid=3437953&group_id=16077&atid=116077

From upstream SVN:
http://freeassociation.svn.sourceforge.net/viewvc/freeassociation?view=revision&revision=1042

--- src/libical/icalerror.h.orig	Mon Aug 30 22:05:06 2010
+++ src/libical/icalerror.h	Tue Nov 15 00:49:05 2011
@@ -121,7 +121,11 @@ if(icalerror_get_error_state(x)==ICAL_ERROR_FATAL || \
 void icalerror_set_errno(icalerrorenum x); 
 #endif
 
-#ifdef ICAL_ERRORS_ARE_FATAL
+#ifndef ICAL_ERRORS_ARE_FATAL
+#define ICAL_ERRORS_ARE_FATAL 0
+#endif
+
+#if ICAL_ERRORS_ARE_FATAL == 1
 #undef NDEBUG
 #endif
 
@@ -131,7 +135,7 @@ void icalerror_set_errno(icalerrorenum x); 
 #define icalerror_check_component_type(value,type);
 
 /* Assert with a message */
-#ifdef ICAL_ERRORS_ARE_FATAL
+#if ICAL_ERRORS_ARE_FATAL == 1
 
 #ifdef __GNUC__
 #define icalerror_assert(test,message) if(!(test)){fprintf(stderr,"%s(), %s:%d: %s\n",__FUNCTION__,__FILE__,__LINE__,message);icalerror_stop_here(); abort();}
